sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListener.scala

```diff
@@ -328,7 +340,18 @@ class SQLAppStatusListener(
     }.toSet
     stageMetrics.keySet().asScala
       .filter(!activeStages.contains(_))
-      .foreach(stageMetrics.remove)
+      .foreach { stageId =>
+        stageMetrics.remove(stageId)
+        if (live) {
```

Comment:
Is this new requirement, or existing bug? If that's an existing bug, it would be nice to separate both cases and deal with other (small) PR.

Reply:
Yeah, it's a new requirement.